src/main/java/bot/util/PrettyTime.java

```diff
@@ -0,0 +1,202 @@
+     * Checks whether this PrettyTime has a LocalDate
+     * representation.
+     *
+     * @return True, if there exists the LocalDate
+     *     representation.
+     */
+    public boolean hasLocalDate() {
+        return this.date.isPresent();
+    }
+
+    /**
+     * Checks whether this PrettyTime actually has
+     * a value for time, apart from the date. The time
+     * can only exist if the LocalDate exists
+     *
+     * @return True, if there exists a time (which
+     *     would also imply that the date exists)
+     */
+    public boolean hasTime() {
+        return this.time.isPresent();
+    }
+
+    /**
+     * Generates a String representing this
+     * PrettyTime in the same format in which
+     * it was entered (DD-MM-YYYY-tttt, if
+     * the LocalDate representation exists)
+     *
+     * @return String representing date and time
+     *     in DD-MM-YYYY-tttt format, if possible
+     */
+    public String toRaw() {
+        return this.date.map(
+            date -> date.format(DateTimeFormatter.ofPattern("dd-MM-y"))
+                    + this.time.map(t -> "-" + String.format("%04d", t))
+                    .orElse("")
+        ).orElseGet(
+                () -> this.rawText.orElse("")
+        );
+    }
+
+    @Override
+    public int compareTo(PrettyTime pt) {
+        if (this.hasLocalDate() && pt.hasLocalDate()) {
+            // if date is the same, check the time
+            if (this.date.get().equals(pt.date.get())) {
+                return this.time.get().compareTo(pt.time.get());
+            } else {
+                return this.date.get().compareTo(pt.date.get());
+            }
+        } else if (!this.hasLocalDate() && !pt.hasLocalDate()) {
+            return this.rawText.get().compareTo(pt.rawText.get());
+        } else if (this.hasLocalDate() && !pt.hasLocalDate()) {
+            return -1;
+        } else {
+            // !this.hasLocalDate() && pt.hasLocalDate()
+            return 1;
+        }
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof PrettyTime) {
+            PrettyTime pt = (PrettyTime) o;
+            if (this.hasLocalDate() && pt.hasLocalDate()) {
+                if (this.hasTime() && pt.hasTime()) {
+                    return this.date.get().equals(pt.date.get())
+                            && this.time.get().equals(pt.time.get());
+                } else if (!this.hasLocalDate() && !pt.hasLocalDate()) {
+                    return this.date.get().equals(pt.date.get());
+                } else {
+                    return false;
+                }
+            } else if (!this.hasLocalDate() && !pt.hasLocalDate()) {
+                return this.rawText.get().equals(pt.rawText.get());
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks whether this PrettyTime has the same
+     * date as another PrettyTime. Time is ignored
```

Comment:
Try to punctuate the end of the sentence of a JavaDoc comment; "Time is ignored."

Reply:
Thank you for the review!!